### PR TITLE
Support prefersEphemeralWebBrowserSession and useWebView authorize options

### PR DESCRIPTION
**Pull request by carmenlau**
ref #73

### CODE REVIEW

File: Sources/Authgear.swift [hunk 1]
```diff
@@ -199,6 +202,9 @@ public class Authgear: NSObject {
 
     public private(set) var sessionState: SessionState = .unknown
 
+    private var currentWebViewRedirectURI: String?
+    public private(set) var authorizeRedirectionHandler: AuthorizeRedirectionHandler = { _ in }
```
Comment (louischan-oursky) [inline Sources/Authgear.swift:206]: Can it be private?

File: Sources/Authgear.swift [hunk 2]
```diff
@@ -7,6 +7,7 @@ import WebKit
 
 public typealias AuthorizeCompletionHandler = (Result<AuthorizeResult, Error>) -> Void
 public typealias ReauthenticateCompletionHandler = (Result<ReauthenticateResult, Error>) -> Void
+public typealias AuthorizeRedirectionHandler = (URL) -> Void
```
Comment (louischan-oursky) [inline Sources/Authgear.swift:10]: Can it be private?

File: Sources/Authgear.swift [hunk 3]
```diff
@@ -615,17 +743,22 @@ public class Authgear: NSObject {
...
     ) {
-        let handler = self.withMainQueueHandler(handler)
-        self.workerQueue.async {
-            self.authorizeWithSession(options, handler: handler)
+        if options.useWebView == true {
+            self.authorizeWithoutSession(options, handler: handler)
+        } else {
+            let handler = self.withMainQueueHandler(handler)
+            self.workerQueue.async {
+                self.authorizeWithSession(options, handler: handler)
```
Comment (louischan-oursky) [inline Sources/Authgear.swift:760]: The suffix WithSession and WithoutSession is not too self-explanatory. I suggest we rename them to `authorizeWithWKWebView` and `authorizeWithASWebAuthenticationSession`. Apply to `reauthenticateXXX` as well.

**Review by louischan-oursky — COMMENTED** [review]
Can we emulate prefersEphemeralWebBrowserSession for WKWebView? I tested the following scenario.

1. Login with Facebook Login
2. Logout
3. Login with use another account.
4. Click Facebook Login. This time facebook recognized me as logged in already. I cannot use another account :(

**Comment by carmenlau** [comment]
Updates:
- Updated to support `preferPrivateWebBrowserSession` for WKWebView and added the option to the demo app
- changed `preferPrivateWebBrowserSession` and `useWebView` default to `false`

**Comment by carmenlau** [comment]
Updated in #75

**Pull request closed without merging**